Silicon/CoffeelakePkg/Library/FirmwareUpdateLib/FirmwareUpdateLib.c

```diff
@@ -289,89 +289,126 @@ GetFirmwareUpdateInfo (
   UINT32                        TopSwapRegionOffset;
   UINT32                        RedundantRegionOffset;
   UINT32                        NonRedundantRegionOffset;
+  UINT32                        CompBase;
+  UINT32                        CompSize;
   FIRMWARE_UPDATE_PARTITION     *UpdatePartition;
   FIRMWARE_UPDATE_REGION        *UpdateRegion;
-  UINT8                          Idx;
+  UINT8                         Idx;
   FLASH_MAP                     *FlashMap;
+  BOOLEAN                       IsBackup;
+  EFI_STATUS                    Status;
 
   TopSwapRegionOffset       = 0;
   RedundantRegionOffset     = 0;
   NonRedundantRegionOffset  = 0;
 
-  //
-  // Get Region sizes from flash map
-  //
-  GetRegionInfo (&TopSwapRegionSize, &RedundantRegionSize, &NonRedundantRegionSize);
+  FlashMap = GetFlashMapPtr();
+  if (FlashMap == NULL) {
+    return EFI_NOT_FOUND;
+  }
 
   AllocateSize    = sizeof (FIRMWARE_UPDATE_PARTITION) + (MAX_UPDATE_REGIONS - 1) * sizeof(FIRMWARE_UPDATE_REGION);
   UpdatePartition = (FIRMWARE_UPDATE_PARTITION *) AllocateZeroPool(AllocateSize);
   ASSERT (UpdatePartition != NULL);
 
   UpdatePartition->RegionCount   = 0;
 
-  // BIOS region layout is as below.
-  //
-  //  +-------------------------+
-  //  +        Top Swap         +
-  //  +-------------------------+
-  //  +     Top Swap Backup     +
-  //  +-------------------------+
-  //  +    Redundant Region     +
-  //  +-------------------------+
-  //  + Redundant Region Backup +
-  //  +-------------------------+
-  //  +  Non Redundant Region   +
-  //  +-------------------------+
-
-  FlashMap = GetFlashMapPtr();
-  if (FlashMap == NULL) {
-    return EFI_NOT_FOUND;
-  }
-
-  NonRedundantRegionOffset = \
-                             FlashMap->RomSize - (TopSwapRegionSize + RedundantRegionSize) * 2 - NonRedundantRegionSize;
+  if (CompareGuid(&ImageHdr->UpdateImageTypeId, &gSblCompFWUpdateImageFileGuid) == TRUE) {

```

Comment:
it is not necessary to compare with TRUE since Compare() itself is a BOOLEAN.

Reply:
Done